pype/hosts/tvpaint/__init__.py

```diff
@@ -31,6 +33,44 @@ def on_instance_toggle(instance, old_value, new_value):
         pipeline._write_instances(current_instances)
 
 
+def application_launch():
+    # Setup project settings if its the template that's launched.
+    if "PYPE_TVPAINT_LAUNCHED_TEMPLATE_FILE" in os.environ:
```

Comment:
1.) This can be function in lib (like `set_context_settings` or something like that) so it's callable from different places. (Like when button is clicked - future).
    - function should expect asset document as an argument because in TVPaint it is possible to have opened multiple workfiles with different asset in context at the same time
2.) Be more specific in environment condition and rather skip unnecessary indentation block of whole function with oposite logic:
```
if os.environ.get("PYPE_TVPAINT_LAUNCHED_TEMPLATE_FILE") != "1":
    return
...
```
3.) Frame start/end values are not checked.

NOTE:
Resolution, frame start/end and handle start/end are always a pair of data. If `data.get("width")` is None but `data.get("height")` is 1000, the height should not be used (happened few times).

Reply:
Good points. Done.